### PR TITLE
[DM-3800][OPCUA] Fix local db memory issue

### PR DESCRIPTION
**Pull request by sourajitsaha17**

### CODE REVIEW

File: content/change-logs/device-management/opcua-10-18-524-0-fix-local-database-memory-issue.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+component:
+  - value: component-Tf05_KQ-B
+    label: OPC UA
+build_artifact:
+  - value: tc-MLn0oFRX-
+    label: opcua
+ticket: DM-3800
+version: 10.18.524.0
+---
+The OPC UA gateway creates a local database namely 'cumulocity-opcua-gateway.db' to store essential data, including a list of executed operation IDs.
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-18-524-0-fix-local-database-memory-issue.md:17]:
```suggestion
The OPC UA gateway creates a local database `cumulocity-opcua-gateway.db` to store essential data, including a list of executed operation IDs.
```

File: content/change-logs/device-management/opcua-10-18-524-0-fix-local-database-memory-issue.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+date:
+title: Fixed local database memory issue for OPC UA gateway
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-18-524-0-fix-local-database-memory-issue.md:3]:
```suggestion
title: New cleanup process during the gateway startup to reduce the size of the database file
```

Comment (sourajitsaha17) [reply content/change-logs/device-management/opcua-10-18-524-0-fix-local-database-memory-issue.md]: @BeateRixen Thanks for the suggestion. After reading the title I realized that I missed a crucial part of the description...that it is a one-time job. So I updated both the description and title.

File: content/change-logs/device-management/opcua-10-18-524-0-fix-local-database-memory-issue.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  - value: component-Tf05_KQ-B
+    label: OPC UA
+build_artifact:
+  - value: tc-MLn0oFRX-
+    label: opcua
+ticket: DM-3800
+version: 10.18.524.0
+---
+The OPC UA gateway creates a local database `cumulocity-opcua-gateway.db` to store essential data, including a list of executed operation IDs.
+As this list expands, the database file consumes more disk space. This issue has been resolved with the latest update by introducing an in-memory cache to store executed operation IDs with each entry set auto-expire after 24 hours.
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-18-524-0-fix-local-database-memory-issue.md:18]:
```suggestion
As this list expands, the database file consumes more disk space. This issue has been resolved by introducing an in-memory cache to store executed operation IDs with each entry set to auto-expire after 24 hours. 
```

**Review by BeateRixen — APPROVED** [review]

**Review by Oezge-Akin — APPROVED** [review]

**Review by eliasweingaertner — APPROVED** [review]

**Pull request merged**